src/pages/iou/IOUCurrencySelection.js

```diff
@@ -0,0 +1,237 @@
+    currencyList: {},
+};
+
+class IOUCurrencySelection extends Component {
+    constructor(props) {
+        super(props);
+
+        const {currencyOptions} = getCurrencyListForSections(this.props.currencyList,
+            '');
+
+        this.state = {
+            searchValue: '',
+            currencyData: currencyOptions,
+            selectedCurrency: {
+                currencyCode: this.props.myPersonalDetails.preferredCurrencyCode,
+                currencySymbol: this.props.myPersonalDetails.preferredCurrencySymbol,
+            },
+        };
+        this.toggleOption = this.toggleOption.bind(this);
+        this.renderItem = this.renderItem.bind(this);
+        this.getSections = this.getSections.bind(this);
+    }
+
+    componentDidMount() {
+        fetchCurrencyPreferences(true);
+        getCurrencyList();
+    }
+
+    /**
+     * Returns the sections needed for the OptionsSelector
+     *
+     * @param {Boolean} maxParticipantsReached
+     * @returns {Array}
+     */
+    getSections() {
+        const sections = [];
+
+        sections.push({
+            title: 'ALL CURRENCIES',
+            data: this.state.currencyData,
+            shouldShow: true,
+            indexOffset: 0,
+        });
+
+        return sections;
+    }
+
+    /**
+     * Returns the key used by the list
+     * @param {Object} option
+     * @return {String}
+     */
+    extractKey(option) {
+        return option.currencyCode;
+    }
+
+    /**
+     * Function which renders a row in the list
+     *
+     * @param {String} currencyCode
+     *
+     */
+    toggleOption(currencyCode) {
+        this.setState({
+            selectedCurrency: {
+                currencyCode,
+                currencySymbol: this.props.currencyList[currencyCode].symbol,
+            },
+        });
+    }
+
+    /**
+     * Function which renders a row in the list
+     *
+     * @param {Object} params
+     * @param {Object} params.item
+     *
+     * @return {Component}
+     */
+    renderItem({item, key}) {
+        return (
+            <OptionRow
+                key={key}
+                mode="compact"
+                option={item}
+                onSelectRow={() => this.toggleOption(item.currencyCode)}
+                isSelected={item.currencyCode === this.state.selectedCurrency.currencyCode}
+                showSelectedState
+                hideAdditionalOptionStates
+            />
+        );
+    }
+
+    /**
+     * Function which renders a section header component
+     *
+     * @param {Object} params
+     * @param {Object} params.section
+     * @param {String} params.section.title
+     *
+     * @return {Component}
+     */
+    renderSectionHeader({section: {title}}) {
+        return (
+            <View>
+                <Text style={[styles.p5, styles.textMicroBold, styles.colorHeading]}>
+                    {title}
+                </Text>
+            </View>
+        );
+    }
+
```

Comment:
I appreciate the separation, but we prefer to keep all render code within the single render block. If this is reusable it could be separated into a new component, but else could you group it with the rest of the render code please.

Reply:
+1

Reply:
Done!

Reply:
@jasperhuangg  this was where I made the renderSectionHeader change. Perhaps I should do it for renderItem as well!